packages/terra-application-layout/src/ApplicationLayout.jsx

```diff
@@ -0,0 +1,222 @@
+      '/': {
+        path: '/',
+        component: {
+          tiny: componentConfig,
+          small: componentConfig,
+        },
+      },
+    };
+  }
+
+  /**
+   * Builds and returns the routing configuration object for all menus with ApplicationMenuConfigAdapters
+   * wrapped around each component entry.
+   */
+  static buildWrappedMenuConfig(props, menuConfig) {
+    const { nameConfig, utilityConfig, extensions } = props;
+
+    const updatedMenuConfig = {};
+    Object.keys(menuConfig).forEach((menuKey) => {
+      const newMenu = Object.assign({}, menuConfig[menuKey]);
+
+      const newMenuComponent = Object.assign({}, newMenu.component);
+
+      navigationLayoutSizes.forEach((size) => {
+        if (!newMenuComponent[size]) {
+          return;
+        }
+
+        const configForSize = Object.assign({}, newMenuComponent[size]);
+
+        const propsForSize = Object.assign({}, configForSize.props);
+
+        propsForSize.applicationMenuConfigAdapterProps = {
+          overrideComponentClass: configForSize.componentClass,
+          nameConfig,
+          utilityConfig,
+          extensions,
+        };
+        configForSize.props = propsForSize;
+        configForSize.componentClass = ApplicationMenuConfigAdapter;
+
+        newMenuComponent[size] = configForSize;
+      });
+
+      newMenu.component = newMenuComponent;
+      updatedMenuConfig[menuKey] = newMenu;
+    });
+
+    return updatedMenuConfig;
+  }
+
+  /**
+   * Builds and returns the routing configuration object for the ApplicationLayout by injecting the PrimaryNavigationSideMenu
+   * and ApplicationMenuConfigAdapters as necessary.
+   */
+  static buildRoutingConfig(props) {
+    const { routingConfig } = props;
+
+    const updatedConfig = Object.assign({}, routingConfig, {
+      menu: ApplicationLayout.buildWrappedMenuConfig(props, Object.assign({}, routingConfig.menu, ApplicationLayout.buildNavigationMenuConfig(props))),
+    });
+
+    return updatedConfig;
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      applicationLayoutRoutingConfig: ApplicationLayout.buildRoutingConfig(this.props),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.nameConfig !== nextProps.nameConfig ||
+        this.props.utilityConfig !== nextProps.utilityConfig ||
+        this.props.routingConfig !== nextProps.routingConfig ||
+        this.props.navigationItems !== nextProps.navigationItems ||
+        this.props.indexPath !== nextProps.indexPath) {
+      this.setState({
+        applicationLayoutRoutingConfig: ApplicationLayout.buildRoutingConfig(nextProps),
+      });
+    }
+  }
+
+  render() {
+    const { app, nameConfig, utilityConfig, navigationItems, indexPath, extensions } = this.props;
+    const { applicationLayoutRoutingConfig } = this.state;
+
+    return (
+      <NavigationLayout
+        app={app}
+        config={applicationLayoutRoutingConfig}
+        header={(
+          <ApplicationHeader
```

Comment:
Where are we toggling the `ApplicationHeader` and `ApplicationMenu` renderings?

Reply:
Are you asking where we're flexing what the menu/header are rendering based on size? The _ApplicationMenu and _ApplicationHeader component handle that flexing internally.

Reply:
Where are we creating the ApplicaitonMenu then? I think I missed it

Reply:
Talked offline and this is done in `buildRoutingConfig()`